### PR TITLE
strip unused audio settings, hide the rest

### PR DESCRIPTION
**Pull request by zzmp**
- remove parameters on the jitter buffer algorithm, these are now static
- remove parameters on the output starve detection algorithm, now static
- move interface audio settings to the Developer menu

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: e2ff738fc3e37b431046de65f6adb93a97336541 available at: https://gist.github.com/6095e2dad71b1781e6b3096e12a1f1ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-e2ff738.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-e2ff738.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 81b6bb3a2dc81f3a082b3f9add3f72996d7dbbd0 available at: https://gist.github.com/6095e2dad71b1781e6b3096e12a1f1ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-81b6bb3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-81b6bb3.dmg

**Comment by ZappoMan** [comment]
Test plan:
- Smoke test
- focus on making sure audio works correctly

File: assignment-client/src/audio/AudioMixer.cpp [hunk 1]
```diff
@@ -899,66 +898,66 @@ void AudioMixer::broadcastMixes() {
 void AudioMixer::parseSettingsObject(const QJsonObject &settingsObject) {
     if (settingsObject.contains(AUDIO_BUFFER_GROUP_KEY)) {
         QJsonObject audioBufferGroupObject = settingsObject[AUDIO_BUFFER_GROUP_KEY].toObject();
+        qDebug() << audioBufferGroupObject;
```
Comment (ZappoMan) [inline assignment-client/src/audio/AudioMixer.cpp]: excessive logging?

File: libraries/audio/src/InboundAudioStream.h [hunk 2]
```diff
@@ -225,60 +145,45 @@ public slots:
 
     AudioRingBuffer _ringBuffer;
 
-    bool _lastPopSucceeded;
+    bool _lastPopSucceeded{ false };
```
Comment (ZappoMan) [inline libraries/audio/src/InboundAudioStream.h]: space between

File: libraries/audio/src/InboundAudioStream.h [hunk 3]
```diff
@@ -225,60 +145,45 @@ public slots:
...
     AudioRingBuffer::ConstIterator _lastPopOutput;
     
-    bool _dynamicJitterBuffers;         // if false, _desiredJitterBufferFrames is locked at 1 (old behavior)
-    int _staticDesiredJitterBufferFrames;
-
-    // if jitter buffer is dynamic, this determines what method of calculating _desiredJitterBufferFrames
-    // if true, Philip's timegap std dev calculation is used.  Otherwise, Freddy's max timegap calculation is used
-    bool _useStDevForJitterCalc;
-
+    bool _dynamicJitterBufferEnabled{ DEFAULT_DYNAMIC_JITTER_BUFFER_ENABLED };
```
Comment (ZappoMan) [inline libraries/audio/src/InboundAudioStream.h]: space between

File: libraries/audio/src/InboundAudioStream.h [hunk 4]
```diff
@@ -225,60 +145,45 @@ public slots:
...
+    int _staticJitterBufferFrames{ DEFAULT_STATIC_JITTER_FRAMES };
```
Comment (ZappoMan) [inline libraries/audio/src/InboundAudioStream.h]: space between

File: libraries/audio/src/InboundAudioStream.h [hunk 5]
```diff
@@ -225,60 +145,45 @@ public slots:
...
+    bool _isStarved{ true };
+    bool _hasStarted{ false };
 
     // stats
 
-    int _consecutiveNotMixedCount;
-    int _starveCount;
-    int _silentFramesDropped;
-    int _oldFramesDropped;
+    int _consecutiveNotMixedCount{ 0 };
```
Comment (ZappoMan) [inline libraries/audio/src/InboundAudioStream.h]: space between

**Comment by kevinmthomasse** [comment]
testing

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** FAILURE
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** FAILURE
**QA Pass**: **QA Pass** FAIL (due to not being able to connect to ```dev-playa``` and ```dev-demo```
**Additional Test Results + Observations**: Unable to connect to ```dev-playa``` and ```dev-demo``` as protocol mismatch. Audio worked perfectly on ```localhost``` no issue. Pressure tested as well for sustainability of tones with reverb no issues either.

**Comment by zzmp** [comment]
@kevinmthomasse @cozza13 is this a pass or a fail?
Do I need to pull from master to account for the protocal mismatch, even though there are no conflicts?

**Comment by cozza13** [comment]
@zzmp please update from master

**Comment by hifi-gustavo** [comment]
Build console output for commit: 09a96796de83347256ebeea3bd5641595d7987c2 available at: https://gist.github.com/6095e2dad71b1781e6b3096e12a1f1ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-09a9679.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-09a9679.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a9318d052bb552041285498e5e672998f0cc6361 available at: https://gist.github.com/6095e2dad71b1781e6b3096e12a1f1ab<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-a9318d0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8636-a9318d0.dmg

**Comment by kevinmthomasse** [comment]
## QA Test
**QA Tester**: KevinMThomas
**localhost 'createTestBlocks.js' Result**: **localhost 'createTestBlocks.js'** SUCCESS
**localhost ATP Upload + Edit Result**: **localhost ATP** SUCCESS
**Echo Server Audio**: **Echo Server Audio** SUCCESS
**localhost 'harmonicOscillator.js' Result**: **localhost 'harmonicOscillator.js'** SUCCESS
**localhost 'reverbTest.js' Result**: **localhost 'reverbTest.js'** SUCCESS
**localhost 'playaPerformanceTest.js' Result**: **localhost 'playaPerformanceTest.js'** SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: **dev-playa 'playaPerformanceTest.js'** SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: **dev-demo 'playaPerformanceTest.js'** SUCCESS
**QA Pass**: **QA Pass** PASS
**Additional Test Results + Observations**: No issues of any kind connecting to ```dev-playa``` or ```dev-demo```. Audio worked seamlessly no dropouts of any kind HOWEVER when doing a rapid teleport from domains I crashed.  I completely retested again and I did NOT crash from rapid teleport as it must have been an isolated unrelated issue as I COULD NOT replicate the crash.


**Domain** |localhost |dev-playa |dev-demo
---|---|---|---
**Render Rate**: |60 |60 |60
**Present Rate**: |60 |60 |60
**Present New Rate**: |53 |60 |59
**CPU Buffer Count**: |2209 |2644 |1571
**GPU Buffer Count**: |667 |521 |356
**CPU Memory Usage**: |29548544 |43089920 |22917120
**GPU Memory Usage**: |20815872 |18833408 |13176832
**CPU Texture Count**: |190 |166 |187
**GPU Texture Count**: |143 |109 |127
**GPU Transfer Count**: |0 |20 |18
**CPU Texture Memory Usage**: |126991632 |316397792 |339159576
**GPU Texture Memory Usage**: |357040128 |121798656 |113721344
**GPU Virtual Memory Usage**: |613167533|486566212|431140972
**Triangle Count**: |546324 |453321 |204692
**Triangle Rate**: |34145250 |30221400 |12793250
**API Drawcall Count**: |250 |201 |127
**Drawcall Count**: |257 |198 |127
**Drawcall Rate**: |16062 |13200 |7937
**State Changes Texture Count**:|258 |222 |269
**Pipeline Count**: |45 |79 |69
**Item Opaques**: |266 |300 |124
**Item Translucents**: |20 |138 |88
**Item Lights**: |21 |57 |15
**Timing Opaques**: |0.638 |0.984 |0.514
**Timing Translucents**: |0.039 |0.429 |0.273
**Timing Lighting**: |0.105 |0.136 |0.109
**Render Frame**: |1.384 |2.669 |1.839

**Pull request merged**